test/tests/searchTest.js

```diff
@@ -83,4 +83,58 @@ describe("Zotero.Search", function() {
 			assert.propertyVal(conditions[0], 'value', 'foo')
 		});
 	});
+
+	describe("#search()", function () {
+		let win;
+	
+		before(function () {
+			// Hidden browser, which requires a browser window, needed for charset detection
+			// (until we figure out a better way)
+			if (!Zotero.isStandalone) {
+				return loadBrowserWindow().then(window => win = window);
+			}
+		});
+
+		after(function () {
+			if (win) {
+				win.close();
+			}
+		});
+	
+		it('should return results with fulltext conditions', function* () {
+			let id = yield importFromTestFile("search_foo.html");
+			let s = new Zotero.Search();
+			s.addCondition('fulltextWord', 'contains', 'foo');
+			let results = yield s.search();
+			assert.equal(results[0], id);
+		});
+		
+		it('should not return failed matches with fulltext conditions', function* () {
+			let id = yield importFromTestFile("search_foo.html");
+			let s = new Zotero.Search();
+			s.addCondition('fulltextWord', 'contains', 'baz');
+			let results = yield s.search();
+			assert.lengthOf(results, 0);
+		});
+		
+		it('should return results for ALL fulltext conditions', function* () {
+			let id = yield importFromTestFile("search_foobar.html");
+			let s = new Zotero.Search();
+			s.addCondition('joinMode', 'all');
+			s.addCondition('fulltextWord', 'contains', 'foo');
+			s.addCondition('fulltextWord', 'contains', 'bar');
+			let results = yield s.search();
+			assert.equal(results[0], id);
+		});
+		
+		it('should not return failed matches for ALL fulltext conditions', function* () {
```

Comment:
Despite the condition names, let's go with "full-text" (with hyphen) for these, and also say "for full-text conditions in ALL mode" for clarity.